R/blob.R

```diff
@@ -19,48 +27,55 @@ setOldClass("blob")
 #'
 #' as.blob(c("Good morning", "Good evening"))
 blob <- function(...) {
-  new_blob(list(...))
+  x <- list2(...)
+  check_raw_list(x)
```

Comment:
Given this is an assertion about type, shouldn't it occur in the constructor? (And shouldn't it be the responsibility of `list_of()`?) That said, it's potential expensive, so maybe vctrs should implement and export `validate_list_of()`?

Reply:
Implemented and exported `validate_blob()` here, opened an issue in _vctrs_. Validation occurs in the constructor for back-compat reasons.